### PR TITLE
[YUNIKORN-2792] Create design doc

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Add the design doc for non-Yunikorn pod tracking.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [x] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2792

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: docs/design/foreign_pod_tracking.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+## Introduction
+
+Currently, we only know about pods in the scheduler core which have been scheduled by Yunikorn. Resource calculation which decides whether a pod fits a node or not depends on a field called `occupiedResources`. This is first calculated in the shim then sent to the core.
```
Comment (chenyulin0719) [inline docs/design/foreign_pod_tracking.md:27]: Yunikorn -> YuniKorn

File: docs/design/foreign_pod_tracking.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+In Yunikorn 1.7, the data in the REST output will not change. We can still calculate
+OccupiedResources information per node. We also announce that AllocatedResources is
+about to change, because the value will be based on all pods on the node.
+
+From Yunikorn 1.8 onwards, we remove OccupiedResources and update the calculation of
+`AllocatedResources`.
+
+To support easier integration with the web UI, a separate list will be maintained for the two
+types of allocations. Allocations made by Yunikorn are still available under `allocations`.
+Non-Yunikorn allocations will be listed under `foreign_allocations`.
```
Comment (chenyulin0719) [inline docs/design/foreign_pod_tracking.md:205]: Besides allocations, we're also tracing container count metrics too(allocated, released, rejected)(YuniKorn only). Should we also trace non-YuniKorn container count? 

In the front page of Web UI, we called below API to get totalContainers count:
- http://localhost:9889/ws/v1/history/containers

The totalContainers was calculated by "totalContainersRunning := allocatedContainers(YuniKorn) - releasedContainers(YuniKorn)

Comment (chenyulin0719) [reply docs/design/foreign_pod_tracking.md:205]: > Should we also trace non-YuniKorn container count?

After rethink it, I believe we don't need to trace it.
Knowning the total non-YuniKorn container count doesn't help us much. 
I only want to know how many running containers are scheduled by YuniKorn in all applications.

Comment (craigcondit) [inline docs/design/foreign_pod_tracking.md:205]: Nit: Let's not use underscores in fields: `foreignAllocations` vs. `foreign_allocations`.

**Review by craigcondit — CHANGES_REQUESTED** [review]
Looks good for the most part. Minor nit, plus I agree with @chenyulin0719 regarding Yunikorn -> YuniKorn.

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**